src/DnsClient/DnsQueryExtensions.cs

```diff
@@ -618,14 +618,14 @@ public static async Task<ServiceHostEntry[]> ResolveServiceAsync(this IDnsQuery
             return ResolveServiceProcessResult(result);
         }
 
-        private static string ConcatResolveServiceName(string baseDomain, string serviceName, string tag)
+        public static string ConcatResolveServiceName(string baseDomain, string serviceName, string tag)

```

Comment:
I'd rather not make those public. Not sure if this was intentional, can you explain what you did here? ;)
Especially `ResolveServiceProcessResult` is very specific and works only with SRV query results.

Reply:
The reason was that I need to lookup SRV records in the cache. To do that I need to construct the hostname query string. And since you already had those methods I thought it'd be better to use them and avoid duplication. But it's a small thing since it's just a string concatenation. If you'd still prefer to make them private I'm happy to revert?